src/main/java/dev/slimevr/protocol/DataFeedBuilder.java

```diff
@@ -242,4 +245,44 @@ public static int createDevicesData(
 
 		return DataFeedUpdate.createDevicesVector(fbb, devicesDataOffsets);
 	}
+
+	public static int createBonesData(
+		FlatBufferBuilder fbb,
+		boolean shouldSend,
+		List<BoneInfo> boneInfos
+	) {
+		if (!shouldSend) {
+			return 0;
+		}
+
+		var boneOffsets = new int[boneInfos.size()];
+		for (var i = 0; i < boneInfos.size(); ++i) {
+			var bi = boneInfos.get(i);
+
+			var rotG = bi.node.worldTransform.getRotation();
+
+			// TODO: The protocol ought to use the tail bone instead of the head
+			// bone, but since I already made the change I'm going to fix that before
+			// the final review.
+			var headPosG = bi.node.getParent().worldTransform.getTranslation();
+			// TODO: make this value not be stale, so that we don't need to
```

Comment:
This gets called when offsets are updated, idk why you need to call it here, better investigate why it's not updated properly than adding it

Reply:
See here: https://discord.com/channels/817184208525983775/932367304233201684/988273403842666537

The TLDR is that it *wasnt* updated - this fixes that by ensuring we *always* update. I would like to fix the root problem, but this ensures that I can be confident in the short term that the lengths reflect the local transforms.